### ISSUE TITLE
The need for 3D objects loader

### ISSUE THREAD

**Issue by ahmedaomda** [javafxports/openjdk-jfx#203]
I have been searching around for a 3D loader for any of the major 3D Models extensions such as FBX I think it's popular request to have a FBX loader in javafx which is a well known extension used by most of  the 3D Modeling tool such as Maya and 3D Max and also used in Unity which means a lot of assets .

**Comment by nlisker** [comment]
This is [JDK-8091851](https://bugs.openjdk.java.net/browse/JDK-8091851). I agree that there is a need for it (several questions on SO as well).

There is a 3rd party [FBX importer](https://github.com/tesis-dynaware/fbx-importer), but it is limited in what it imports. There is also the 3D viewer in the [repo](http://hg.openjdk.java.net/openjfx/jfx-dev/rt/file/c263b8ea1909/apps/samples/3DViewer), which isn't being maintained and supports several importers, with .ma being the only one capable of animations.

The question is: who will be willing to work on it?

**Comment by nlisker** [comment]
So is it possible to work on this under apps/samples?

**Comment by nlisker** [comment]
@jperedadnr Would you be interested in working on 3D importers in some fashion?

**Comment by jperedadnr** [comment]
@nlisker Definitely I'm interested. 

As Kevin said, I don't think this should be part of the JavaFX core, and I wouldn't have them hidden in a sample, even if 3D Viewer is _the_ reference for JavaFX 3D.

I'd say that we could add the existing importers from the [3DViewer](
https://github.com/javafxports/openjdk-jfx/tree/develop/apps/samples/3DViewer/src/main/java/com/javafx/experiments/importers) project to [FXyz 3D](https://github.com/FXyz/FXyz), creating a new artifact (org.fxyz3d:fxyz3d-importers or alike). 

As far as I know these importers are BSD3 licensed, so that wouldn't be a problem. Anyone could work on improving them if possible, and working on new ones (FBX, ...).

Let's see what others say about this?

**Comment by johanvos** [comment]
I agree 3D importers are very valuable. While we can have a sample that both contains the importer as well as an example on how to use it, it makes much more sense to me to have those importers (and more code) in a third-party project (e.g. FXyz 3D) if the maintainers of that project want to do that.

Still, there needs to be a sample, but that then belongs in the FXyz samples (which depends on OpenJFX but not the other way round).

In general, I think this would be a great example on how third party projects extend the OpenJFX core modules.

**Comment by nlisker** [comment]
I don't mind in which code base they are (FXyz is fine), I just want them to exist and work properly, and am ready to contribute.

I also don't know about the licensing, but I doubt it would be a problem.

**Comment by Birdasaur** [comment]
I think this move is a good one and sensible. FX 3D is useful now but improving access to 3D models would dramatically increase that usefulness.  Consolidating these importers with FXyz is a reasonable move that will help "lower the bar" of difficulty for Java developers to use JavaFX 3D for rapid development.  (including AR support in JavaFX which I dream for and could be a killer feature that I believe would elevate JavaFX interest significantly )

I am in agreement with nlisker in that I do not personally care which code base they exist in.  We only started this project years ago to make it easier for other developers to use FX 3D and that is what I want today.  This consolidation is something that should have been done 4 years ago honestly.

**Comment by shathor** [comment]
Until this is realized there are a couple of importers by InteractiveMesh: http://www.interactivemesh.org/models/jfx3dimporter.html. Supported formats: 3ds, dae, zae, fxml, obj, stl, x3d, x3dz. I'm using it for obj and it works pretty well. **However its licencing can be problematic.**

**Comment by omega09** [comment]
I use the 3DViewer for obj and it also works fine. The issue is with the more complex types that have animation, materials and other data.

**Comment by jperedadnr** [comment]
FYI, I've started the work on FXyz-Importers with a first [PR](https://github.com/FXyz/FXyz/pull/42), anyone is welcome to check and review.
